app/recipe/tests/test_ingredients_api.py

```diff
@@ -0,0 +1,104 @@
+"""
+Tests for the Ingredients API.
+"""
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from django.test import TestCase
+
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core.models import (
+    Ingredient,
+)
+
+from recipe.serializers import IngredientSerializer
+
+
+INGREDIENTS_URL = reverse('recipe:ingredient-list')
+
+
+def detail_url(ingredient_id):
+    """Create and return an ingredient detail url."""
+    return reverse('recipe:ingredient-detail', args=[ingredient_id])
+
+
+def create_user(email='user@example.com', password='testpass123'):
+    """Create and return a new user."""
+    return get_user_model().objects.create_user(email, password)
+
+
+class PublicIngredientsApiTests(TestCase):
+    """Test unauthenticated API requests."""
+
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        """Test auth is required for retrieving Ingredients"""
+        res = self.client.get(INGREDIENTS_URL)
+
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class PrivateIngredientsApiTests(TestCase):
+    """Test authenticated API requests."""
```

Comment:
The API specification doesn't describe any separate Ingredients API. All ingredients are created or deleted together with their recipes. Do you think this extra API is useful?